llvm/test/CodeGen/NVPTX/mulhi-intrins.ll

```diff
@@ -0,0 +1,103 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: llc < %s -mtriple=nvptx64 -mcpu=sm_50 | FileCheck %s
```

Comment:
Another improvent for the tests is to add a conditional ptxas compilation run and then actually run LLVM tests with ` LLVM_PTXAS_EXECUTABLE=/path/to/cuda/bin/ptxas` to make sure ptxas does accept the instructions we generate.